themes/article/app/routes/($a).($b).($c).($d).$slug[.json].tsx

```diff
@@ -13,23 +13,27 @@ function api404(message = 'No API route found at this URL') {
 }
 
 export const loader: LoaderFunction = async ({ request, params }) => {
-  const { slug } = params;
+  const [first, ...rest] = new URL(request.url).pathname
+    .slice(1)
+    .replace(/\.json$/, '')
+    .split('/');
   // Handle /myst.xref.json as slug
-  if (slug === 'myst.xref') {
+  if (rest.length === 0 && first === 'myst.xref') {
     const xref = await getMystXrefJson();
     if (!xref) {
       return json({ message: 'myst.xref.json not found', status: 404 }, { status: 404 });
     }
     return json(xref);
   }
   // Handle /myst.search.json as slug
-  else if (slug === 'myst.search') {
+  else if (rest.length === 0 && first === 'myst.search') {
     const search = await getMystSearchJson();
     if (!search) {
       return json({ message: 'myst.search.json not found', status: 404 }, { status: 404 });
     }
     return json(search);
   }
+  const slug = [first, ...rest].join('.');
   const data = await getPage(request, { slug }).catch(() => null);
```

Comment:
(Look at this for example - on this route, we do not mention anything about `project`. The other route should probably be the same way.)